components/style/gecko/wrapper.rs

```diff
@@ -96,10 +92,36 @@ fn from_opaque_style_data(d: *mut OpaqueStyleData) -> *mut NonOpaqueStyleData {
 pub struct GeckoNode<'ln>(pub &'ln RawGeckoNode);
 
 impl<'ln> GeckoNode<'ln> {
+    fn from_content(content: &'ln nsIContent) -> Self {
+        use std::mem;
+        GeckoNode(unsafe { mem::transmute(content) })
+    }
+
+    fn node_info(&self) -> *mut structs::NodeInfo {
+        debug_assert!(!self.0.mNodeInfo.mRawPtr.is_null());
+        self.0.mNodeInfo.mRawPtr
+    }
+
+    fn flags(&self) -> u32 {
+        (self.0)._base._base_1.mFlags
+    }
+
+    // FIXME: We can implement this without OOL calls, but we can't easily given
+    // GeckoNode is a raw reference.
+    //
+    // We can use a Cell<T>, but that's a bit of a pain.
```

Comment:
Pretty much, since the other alternative (implementing it ourselves with `asm!` for platforms we care about) is also unstable.